reverse-proxy-nginx/set-secure-mode.sh

```diff
@@ -0,0 +1,192 @@
+          --publish-add published=80,target=80 \
+          --publish-add published=443,target=443 \
+          instant_$service_name" \
+        throw \
+        "Error updating $service_name service"
+
+    overwrite "Updating $service_name service: adding config for dummy certificates... Done"
+}
+
+function generate_real_certificates() {
+    try "docker run --rm \
+          -p 8083:80 \
+          -p 8443:443 \
+          --name certbot \
+          --network cert-renewal-network \
+          -v data-certbot-conf:/etc/letsencrypt/archive/${DOMAIN_NAME} \
+          certbot/certbot:v1.23.0 certonly -n \
+          --standalone \
+          ${staging_args} \
+          -m ${RENEWAL_EMAIL} \
+          ${domain_args[*]} \
+          --agree-tos" \
+        throw \
+        "Failed to generate certificates"
+
+    try "docker run --rm --network host --name certbot-helper -w /temp \
+          -v data-certbot-conf:/temp-certificates \
+          -v instant:/temp busybox sh \
+          -c \"rm -rf certificates; mkdir -p certificates; cp -r /temp-certificates/* /temp/certificates\"" \
+        throw \
+        "Failed to transfer certificates"
+
+    try "docker volume rm data-certbot-conf" catch "Failed to remove data-certbot-conf volume"
+
+    local new_timestamp
+    new_timestamp="$(date "+%Y%m%d%H%M%S")"
+    create_secrets_from_certificates "${new_timestamp}"
+}
+
+function update_nginx_real_certificates() {
+    local curr_full_chain_name
+    curr_full_chain_name=$(docker service inspect instant_"$service_name" --format "{{(index .Spec.TaskTemplate.ContainerSpec.Secrets 0).SecretName}}")
+    local curr_priv_key_name
+    curr_priv_key_name=$(docker service inspect instant_"$service_name" --format "{{(index .Spec.TaskTemplate.ContainerSpec.Secrets 1).SecretName}}")
+
+    log info "Updating $service_name service: adding secrets for generated certificates..."
+    try "docker service update \
+          --secret-rm ${curr_full_chain_name} \
+          --secret-rm ${curr_priv_key_name} \
+          --secret-add source=${new_timestamp}-fullchain.pem,target=/run/secrets/fullchain.pem \
+          --secret-add source=${new_timestamp}-privkey.pem,target=/run/secrets/privkey.pem \
+          instant_$service_name" \
+        throw \
+        "Error updating $service_name service"
+    overwrite "Updating $service_name service: adding secrets for generated certificates... Done"
+}
+
+function set_nginx_network() {
+    local nginx_network_exists
+    nginx_network_exists=$(docker network ls --filter name=cert-renewal-network --format '{{.Name}}')
+    #Do not create docker network if it exists
+    if [[ -z "${nginx_network_exists}" ]]; then
+        try \
+            "docker network create -d overlay --attachable cert-renewal-network" \
+            throw \
+            "Failed to create cert-renewal-network network"
+    fi
+}
+
+set_secure_mode() {
+    init_vars "$@"
+    import_sources
+
+    local domain_args=()
```

Comment:
`domain_args` should be declared and initialised in `init_vars`

Reply:
nice! thanks done